db/snapshot/backup_task.cc

```diff
@@ -141,28 +141,17 @@ future<> backup_task_impl::do_backup() {
 future<> backup_task_impl::process_snapshot_dir() {
     auto snapshot_dir_lister = directory_lister(_snapshot_dir, lister::dir_entry_types::of<directory_entry_type::regular>());
 
-    for (;;) {
-        std::optional<directory_entry> component_ent;
-        try {
-            component_ent = co_await snapshot_dir_lister.get();
-        } catch (...) {
-            // FIXME: we don't really need this
-            // as process_snapshot_dir is now called sequentially,
-            // before other concurrent errors can happen
-            if (!_ex) {
-                _ex = std::current_exception();
-                break;
-            }
-        }
-        if (!component_ent.has_value()) {
-            break;
+    try {
+        while (auto component_ent = co_await snapshot_dir_lister.get()) {
+            _files.push_back(component_ent->name);
         }
-        _files.push_back(component_ent->name);
+    } catch (...) {
+        _ex = std::current_exception();
     }
 
     co_await snapshot_dir_lister.close();
     if (_ex) {
-        co_await coroutine::return_exception_ptr(std::move(_ex));
+        co_await coroutine::return_exception_ptr(_ex);
```

Comment:
nit: It looks like this change should be merged with the previous patch that introduced the "bug"

Reply:
Can do

Reply:
Squashed with previous patch in v8 ([63026c2](https://github.com/scylladb/scylladb/commit/63026c2151ada0845db2c8201b3367be799a130b))